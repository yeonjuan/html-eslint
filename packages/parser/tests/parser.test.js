"use strict";

const parserForEslint = require("../lib");

describe("parser", () => {
  const parser = {
    parse(code) {
      return parserForEslint.parseForESLint(code).ast;
    },
  };

  it("parse", () => {
    const code = "<html></html>";
    expect(() => parser.parse(code)).not.toThrow();
  });

  it("base", () => {
    const code = `
  <html>
  </html>`;
    expect(parser.parse(code)).toMatchSnapshot();
  });

  it("DOCTYPE", () => {
    const code = `
  <!DOCTYPE HTML>
  <html>
  </html>`;
    expect(parser.parse(code)).toMatchSnapshot();
  });

  it("attr", () => {
    const code = `
<div attr="foo">
</div>
`;
    expect(parser.parse(code)).toMatchSnapshot();
  });

  it("comment", () => {
    const code = `
<div>
<!-- comment -->
</div>
`;
    expect(parser.parse(code)).toMatchSnapshot();
  });

  it("comment multiline", () => {
    const code = `
<!--
  comment
-->
`;
    expect(parser.parse(code)).toMatchSnapshot();
  });

  it("no opening tag", () => {
    const code = `<!DOCTYPE html>
    <html lang="en">
    </head>
    <body>
    
      <table>
        <tr>
          <td>Table 1</td>
        </tr>
      </table>
    
      <table>
        <tr>
          <td>Table 2</td>
        </tr>
      </table>
    
    </body>
    </html>`;
    expect(parser.parse(code)).toMatchSnapshot();
  });
});
