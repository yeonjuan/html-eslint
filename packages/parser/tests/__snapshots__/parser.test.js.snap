// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser base 1`] = `
Object {
  "body": Array [],
  "childNodes": Array [
    Object {
      "attrs": Array [],
      "childNodes": Array [
        Object {
          "attrs": Array [],
          "end": 0,
          "loc": Object {
            "end": Object {
              "column": 0,
              "line": 0,
            },
            "start": Object {
              "column": 0,
              "line": 0,
            },
          },
          "namespaceURI": "http://www.w3.org/1999/xhtml",
          "range": Array [
            0,
            0,
          ],
          "start": 0,
          "tagName": "head",
          "type": "head",
        },
        Object {
          "attrs": Array [],
          "end": 0,
          "loc": Object {
            "end": Object {
              "column": 0,
              "line": 0,
            },
            "start": Object {
              "column": 0,
              "line": 0,
            },
          },
          "namespaceURI": "http://www.w3.org/1999/xhtml",
          "range": Array [
            0,
            0,
          ],
          "start": 0,
          "tagName": "body",
          "type": "body",
        },
      ],
      "end": undefined,
      "loc": Object {
        "end": Object {
          "column": undefined,
          "line": undefined,
        },
        "start": Object {
          "column": undefined,
          "line": undefined,
        },
      },
      "namespaceURI": "http://www.w3.org/1999/xhtml",
      "range": Array [
        undefined,
        undefined,
      ],
      "start": undefined,
      "tagName": "html",
      "type": "html",
    },
  ],
  "comments": Array [],
  "end": undefined,
  "loc": Object {
    "end": Object {
      "column": undefined,
      "line": undefined,
    },
    "start": Object {
      "column": undefined,
      "line": undefined,
    },
  },
  "mode": "quirks",
  "range": Array [
    undefined,
    undefined,
  ],
  "start": undefined,
  "tokens": Array [],
  "type": "Program",
}
`;
